enterprise/backend/test/metabase_enterprise/transforms/search_test.clj

```diff
@@ -0,0 +1,150 @@
+                                                            :target      {:database (mt/id)}
+                                                            :created_at  now
+                                                            :updated_at  now}]
+          (let [ingested-transform (ingest-then-fetch! "transform" "Test Python transform")]
+            (is (=? (index-entity
+                     {:model            "transform"
+                      :model_id         (str transform-id)
+                      :name             "Test Python transform"
+                      :database_id      (mt/id)
+                      :model_created_at now
+                      :model_updated_at now})
+                    ingested-transform))))))
+
+    (testing "A simple MBQL transform gets properly ingested & indexed for search"
+      (let [now (t/truncate-to (t/offset-date-time) :millis)]
+        (mt/with-temp [:model/Transform {transform-id :id} {:name        "Test MBQL transform"
+                                                            :description "An MBQL test transform"
+                                                            :source      {:type "query"
+                                                                          :query (mt/mbql-query venues {:limit 10})}
+                                                            :target      {:database (mt/id)
+                                                                          :table "test_mbql_table"}
+                                                            :created_at  now
+                                                            :updated_at  now}]
+          (let [ingested-transform (ingest-then-fetch! "transform" "Test MBQL transform")]
+            (is (=? (index-entity
+                     {:model            "transform"
+                      :model_id         (str transform-id)
+                      :name             "Test MBQL transform"
+                      :database_id      (mt/id)
+                      :model_created_at now
+                      :model_updated_at now})
+                    ingested-transform))))))))
+
+(deftest transform-query-ingestion-test
+  (testing "Contents of SQL and Python transform sources are extracted and indexed for full-text search"
+    (when (= (mdb/db-type) :postgres)
+      (mt/with-temp [:model/Transform _ {:target {:database (mt/id)
+                                                  :table "test_table"}
+                                         :name "Test SQL transform"
+                                         :source {:type "query"
+                                                  :query {:database (mt/id)
+                                                          :native {:query "SELECT 1"}}}}]
+
+        (let [ingested-transform (ingest-then-fetch! "transform" "Test SQL transform")]
+          (is (= "'1':8B 'select':7B 'sql':2A,5B 'test':1A,4B 'transform':3A,6B"
```

Comment:
Is the format of these guaranteed to be stable? I seem to recall the stemming is somehow dependent on server config, but don't recall details. Probably stable enough / unlikely to change for default postgres images in CI...